src/pages/blog/2024-09-18-codegen.mdx

```diff
@@ -0,0 +1,167 @@
+---
+title: "Generating type safe clients using code generation"
+tags: ["blog"]
+date: 2024-09-18
+byline: Martin Bonnin
+---
+
+A GraphQL endpoint usually returns a JSON payload. While you can use the result as a dynamic object, the GraphQL type system gives us a lot of information about what is inside that JSON payload.
+
+If you're not using code generation, you're missing out on a lot, especially if you're using a type-safe language such as TypeScript, Swift or Kotlin.
+
+By using code generation, you get:
+
+- compile time guarantees about your code and the data it manipulates.
+- Autocomplete and inline documentation in your favorite IDE.
+
+All of that without having to write and maintain types manually!
+
+For simplicity, this post uses TypeScript for code blocks but the same concepts can be applied to Swift/Kotlin.
```

Comment:
Re-reading this, feels like this could be an admonition at the end of the paragraph. Is there anything like this on the blog? Similar to GitHub note?

```markdown
> [!NOTE]  
> For simplicity, this post uses TypeScript for code blocks but the same concepts can be applied to Swift/Kotlin.
```